### DISCUSSION TITLE
Sidecar or not?

### DISCUSSION THREAD

**Discussion by XMethues** [tauri-apps/tauri#4331] [Q&A] [answered]
I'm building an app that rust notify to work some folder and files. After parsing the data in the file(say csv files) send the data to frontend(react) to display it. Since I need to start/stop the process by click button. Should I create a thread for the watch method or a process(sidecar mode)? 
I prefer sidecar mode. If I choose this mode I'm confusing about some concept.
1. The main problem is how to transfer data? Since the watch target is csv files(not to big, some of them only few lines,some of them are hundreds of lines). Creating a simple http/grpc server?
2. if the sidecar failed. Is there a way to emit a notification?

**Comment by simonhyll** [comment, marked as answer]
1. Communicating with the sidecar is gonna have to use something along the lines of what you suggest, http or grpc. If it's enough for  your purposes then I might suggest WebSocket simply because it gives you the option later to connect to it as well with your frontend, in case that's desired. There's no magical IPC with a sidecar, it's literally another binary running on your computer separate from the app, so normal communication limitations apply
2. I guess that depends on how you launch the sidecar. But if you check the documentation you'll find that you can e.g. await the sidecars execution and get its output, you could use that to check for its final status, and you could use the communication in 1) to emit errors and other types of logging <https://tauri.app/v1/guides/building/sidecar>

**Reply by FabianLars** [reply]
Also for simple data like csv, the default stdio communication channel mayyy be enough for you too 🤷
